AK/Error.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Error.h"
+#include "String.h"
+
+namespace AK {
+
+Error::Error(Error&& other)
+{
+    swap(this->m_syscall, other.m_syscall);
+    swap(this->m_code, other.m_code);
+#if !defined(KERNEL)
+    if (other.m_string_literal.has<StringView>())
+        this->m_string_literal = move(other.m_string_literal.get<StringView>());
+    else if (other.m_string_literal.get<String const*>() != nullptr) {
+        this->m_string_literal = move(other.m_string_literal.get<String const*>());
```

Comment:
Both of these `move`s don't do anything and I'd bet clangd even says to remove them

Reply:
clangd actually doesn't say so, I was just trying to be careful about semantics here since we can easily cause memory bugs if we don't do this right.

Reply:
Maybe whatever enables this warning depends on your editor's LSP plugin, but I just checked your branch out and indeed get:

![Screenshot from 2023-02-08 14-54-51](https://user-images.githubusercontent.com/5600524/217637449-7bb10d77-9aef-4c18-9621-327bfa6974b1.png)